src/bun.js/test/jest.zig

```diff
@@ -1384,6 +1384,55 @@ pub const TestScope = struct {
     }
 };
 
+pub const Mock = struct {
+    count: u32,
+
+    pub usingnamespace JSC.Codegen.JSMock;
+
+    pub fn constructor(globalThis: *JSC.JSGlobalObject, _: *JSC.CallFrame) callconv(.C) ?*Mock {
+        var mock = globalThis.allocator().create(Mock) catch unreachable;
+        mock.* = .{
+            .count = 0,
+        };
+        return mock;
+    }
+
+    pub fn fn1(_: *Mock, _: *JSC.JSGlobalObject, _: *JSC.CallFrame) callconv(.C) JSValue {
+        return JSValue.zero;
+    }
+
+    pub const fn2 = NewClass(Mock, .{ .name = "fn" }, .{
```

Comment:
We don't want to use `NewClass` here

`NewClass` is the previous iteration of Bun's JavaScriptCore bindings. That one uses the (slower) JavaScriptCore C API. We haven't migrated all the existing code paths to it, which is why you see it in use still.

`JSC.Codegen.JSMock` is the newer bindings and its the one to use here